Fixes gdist- and pdist_scale node paramter names

DavidPL1 opened the pull request:
Fixes #935. The parameter names are simply changed to fit those defined in the cfg files of BaseLocalPlanner.

mikeferguson commented:
This will break a lot of robot configurations out there. It would seem the proper fix is probably to update the .cfg file to use these names, since that was not working previously anyways.

DavidPL1 commented:
Yes, for those who figured out which parameters to use it would break the configuration. Didn't think about that :D

I guess I could adapt this PR to change the names in the cfg file and try to spread awareness among the movebase users I know. I noticed for instance that the public TIAGo navigation config by PAL Robotics uses the wrong names...

mikeferguson commented:
This does not build - needs changes in the configureCB (see build errors)

DavidPL1 commented:
Yes, I know. Sorry for the delay, had to finish my thesis first :sweat_smile: 
But I'm on it now and running a few tests first and checking for other parameters with the same problem.

DavidPL1 committed:
Fixes BaseLocalPlanner dynamic reconf cfg params

Renames goal and path distance dynamic reconfigure parameter
names in the cfg file in order to actually make the parameters
used by the trajectory planner changeable.

Fixes #935

DavidPL1 committed:
Make goal and path scaling parameter names consistent in dwa and base local planner

This does not change any functionality and is just for better readability.

DavidPL1 commented:
Okay, so the latest commits should do the trick.
Since, if one still uses `pdist_scale` or `gdist_scale` as parameters in their config, they would be overridden by the dynamic reconfigure defaults, the values are copied over to the ones used by dynamic reconfigure (see [here](https://github.com/DavidPL1/navigation/blob/102140c3c76ea399f1bd364eaa4c92b0aa7b32d8/base_local_planner/src/trajectory_planner_ros.cpp#L177)).

In my tests both variants for setting the parameters worked and in both cases they could be changed using dynamic reconfigure.

I also noticed a problem with `max_vel_theta` and `min_vel_theta` when using `max_rotational_vel`: this value is parsed correctly but will instantly be overridden by the dynamic_reconfigure parameter defaults of `min_vel_theta` and `max_vel_theta`, as to dynamic reconfigure they seem to be undefined.
I have not fixed this yet, but this could be done similar to the above solution.

mikeferguson commented on base_local_planner/src/trajectory_planner_ros.cpp:
```diff
@@ -166,8 +166,25 @@ namespace base_local_planner {
+                                                                  0.6);
+      // values of the deprecated params need to be applied to the current params, as defaults 
+      // of defined for dynamic reconfigure will override them otherwise.
+      if (private_nh.hasParam("pdist_scale"))
```
should this logic actually have "& !private_nh.hasParam("path_distance_bias")" -- I presume we only want to do this once

DavidPL1 replied:
Isn't this executed just once? On node restart the path_distance_bias parameter is deleted if it is not set in the config file, so this needs to be done on every node start. Or do you mean something else?

~However, I should add your suggestion for consistency in case of both variables being defined in the config file, because the `loadParameterWithDeprecationWarning` call also favours path_distance_bias over pdist_scale.~
Forget this last part, the param is updated with the correct value regardless if both are defined in the config. So yes, if both are defined the `setParam` call is redundant.

DavidPL1 committed:
Prevent redundant setParam calls for path_distance_bias and goal_distance_bias

as suggested by @mikeferguson
